tests/test_randomizer.py

```diff
@@ -0,0 +1,63 @@
+import pytest
+
+import dice
+import randomizer
+
+
+def test_add_ability_to_pool(ability_d4):
+    randomizer.add_ability_to_pool(ability_d4)
+    assert len(randomizer.Pool) == 1
+
+
+def test_remove_ability_from_pool(ability_d4):
+    initial_len = len(randomizer.Pool)
+
+    randomizer.remove_ability_from_pool(ability_d4)
+    assert len(randomizer.Pool) < initial_len
+
+
+def test_empty_pool(ability_d4, ability_d8):
+    randomizer.add_ability_to_pool(ability_d4)
+    randomizer.add_ability_to_pool(ability_d8)
+
+    randomizer.empty_pool()
+    assert len(randomizer.Pool) == 0
+
+
+def test_pop_value_from_pool(ability_d8):
+    randomizer.add_ability_to_pool(ability_d8)
+
+    pop_value = randomizer.pop_value_from_pool()
+    assert pop_value == ability_d8
+    assert len(randomizer.Pool) == 0
+
+
+def test_get_random_ability_value(ability_d4):
+    random_value = randomizer.get_random_ability_value(ability_d4)
+
+    assert random_value in range(1, 6)
```

Comment:
Oh, this is obvious, don't you see. Ability_d4 says that it is d4, and its values are in [1..4] range, inclusively. And here you state that it can be 5, breaking the assumption.
Moreover, if suddenly your ability_d4 will be reduced to [1..3], you'll not notice it here, because `assert set(range(1, 4)).issubset(set(range(1, 6)))`.